### DISCUSSION TITLE
When using JS modular import, how to pass the return value of JS to Python?

### DISCUSSION THREAD

**Discussion by Autism-al** [zauberzeug/nicegui#3944] [Q&A]
### Question

For example, my JS code has a function that returns an array, and I need to receive this array in Python to complete data processing and display the final result using nicegui. It seems that it cannot pass values in the run_ JavaScript way as described in the document.
```js
export async function generateData(a) {
    try {
        var response = await axios.post(a);
        if (response && response.data) {
            if (response.data.success) {
                var data = JSON.parse(response.data.data);
                return data;
            } else {
                throw response.data.message ?? window.errorAlert;
            }
        }
    } catch (ex) {
        error(ex);
    }
}
```
I imported it into Nicegui：
```ui.add_body_html("""<script type="module" src="js/GetData.js">import {generateData} from "/js/GetData.js";</script>""")```
I would like to know how to get the return value of this function to python. Thank you very much.

**Comment by falkoschindler** [comment]
Hi @Autism-al,

Have you seen https://nicegui.io/documentation/run_javascript#run_javascript? It shows how to await an async JavaScript function.

If this does not help, I kindly ask you to create a minimum reproducible code example so that we can dive into your problem more easily. Thanks!
[Why is it important to provide a minimal reproducible example?](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-it-important-to-provide-a-minimal-reproducible-example)

**Reply by Autism-al** [reply]
Thank you for your help! I have resolved this issue.
